aya/src/maps/mod.rs

```diff
@@ -174,6 +174,10 @@ pub enum MapError {
     #[error(transparent)]
     SyscallError(#[from] SyscallError),
 
+    /// An error indicating that an interior nul byte was found.
+    #[error(transparent)]
+    NulError(#[from] std::ffi::NulError),
```

Comment:
This is not the right way to go. This exposes an implementation detail (we convert to CString), to the aya API.

Reply:
use `#[error("invalid path '{path}' to create CString {source}")]` to the aya API.